examples/failing/UnusableTypeClassMethodSynonym.purs

```diff
@@ -0,0 +1,9 @@
+-- @shouldFailWith UnusableDeclaration
+module Main where
+
+type M x = forall a. a
+
+class C a where
+  -- after synonym expansion, the type doesn't actually contain an `a`
+  c :: M a
```

Comment:
There is a related problem: a type variable can be shadowed:

```purescript
c :: forall a. a -> a
```

has the same problem.